Mathlib/Order/Disjoint.lean

```diff
@@ -0,0 +1,650 @@
+  codisjoint_inf_right.2 ⟨hb, hc⟩
+#align codisjoint.inf_right Codisjoint.inf_right
+
+-- porting note: the `(e :)` trick for elaboration order wasn't needed in Lean 3.
+-- see https://github.com/leanprover/lean4/issues/1892 and
+-- https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/type.20synonym.20trickery.20not.20working
+theorem Codisjoint.left_le_of_le_inf_right (h : a ⊓ b ≤ c) (hd : Codisjoint b c) : a ≤ c :=
+  (@Disjoint.left_le_of_le_sup_right αᵒᵈ _ _ _ _ _ h hd.symm :)
+#align codisjoint.left_le_of_le_inf_right Codisjoint.left_le_of_le_inf_right
+
+theorem Codisjoint.left_le_of_le_inf_left (h : b ⊓ a ≤ c) (hd : Codisjoint b c) : a ≤ c :=
+  hd.left_le_of_le_inf_right <| by rwa [inf_comm]
+#align codisjoint.left_le_of_le_inf_left Codisjoint.left_le_of_le_inf_left
+
+end DistribLatticeTop
+
+end Codisjoint
+
+open OrderDual
+
+theorem Disjoint.dual [SemilatticeInf α] [OrderBot α] {a b : α} :
+    Disjoint a b → Codisjoint (toDual a) (toDual b) :=
+  id
+#align disjoint.dual Disjoint.dual
+
+theorem Codisjoint.dual [SemilatticeSup α] [OrderTop α] {a b : α} :
+    Codisjoint a b → Disjoint (toDual a) (toDual b) :=
+  id
+#align codisjoint.dual Codisjoint.dual
+
+@[simp]
+theorem disjoint_toDual_iff [SemilatticeSup α] [OrderTop α] {a b : α} :
+    Disjoint (toDual a) (toDual b) ↔ Codisjoint a b :=
+  Iff.rfl
+#align disjoint_to_dual_iff disjoint_toDual_iff
+
+@[simp]
+theorem disjoint_ofDual_iff [SemilatticeInf α] [OrderBot α] {a b : αᵒᵈ} :
+    Disjoint (ofDual a) (ofDual b) ↔ Codisjoint a b :=
+  Iff.rfl
+#align disjoint_of_dual_iff disjoint_ofDual_iff
+
+@[simp]
+theorem codisjoint_toDual_iff [SemilatticeInf α] [OrderBot α] {a b : α} :
+    Codisjoint (toDual a) (toDual b) ↔ Disjoint a b :=
+  Iff.rfl
+#align codisjoint_to_dual_iff codisjoint_toDual_iff
+
+@[simp]
+theorem codisjoint_ofDual_iff [SemilatticeSup α] [OrderTop α] {a b : αᵒᵈ} :
+    Codisjoint (ofDual a) (ofDual b) ↔ Disjoint a b :=
+  Iff.rfl
+#align codisjoint_of_dual_iff codisjoint_ofDual_iff
+
+section DistribLattice
+
+variable [DistribLattice α] [BoundedOrder α] {a b c : α}
+
+theorem Disjoint.le_of_codisjoint (hab : Disjoint a b) (hbc : Codisjoint b c) : a ≤ c := by
+  rw [← @inf_top_eq _ _ _ a, ← @bot_sup_eq _ _ _ c, ← hab.eq_bot, ← hbc.eq_top, sup_inf_right]
+  exact inf_le_inf_right _ le_sup_left
+#align disjoint.le_of_codisjoint Disjoint.le_of_codisjoint
+
+end DistribLattice
+
+section IsCompl
+
+/-- Two elements `x` and `y` are complements of each other if `x ⊔ y = ⊤` and `x ⊓ y = ⊥`. -/
+structure IsCompl [PartialOrder α] [BoundedOrder α] (x y : α) : Prop where
+  /-- If `x` and `y` are to be complementary in an order, they should be disjoint. -/
+  protected Disjoint : Disjoint x y
+  /-- If `x` and `y` are to be complementary in an order, they should be codisjoint. -/
+  protected Codisjoint : Codisjoint x y
+#align is_compl IsCompl
+
+theorem IsCompl_iff [PartialOrder α] [BoundedOrder α] {a b : α} :
```

Comment:
Shouldn't the theorems all have `isCompl` rather than `IsCompl`?

Reply:
Yes.

Reply:
Thanks for noticing, I hope I have maintained the correct distinction in the latest commit.